bofire/strategies/predictives/enting.py

```diff
@@ -0,0 +1,131 @@
+    ):
+        super().__init__(data_model=data_model, **kwargs)
+        self._init_problem_config()
+        self._enting = Enting(self._problem_config, data_model.enting_params.dict())
+        self._solver_params = data_model.solver_params
+        self._learn_from_candidates_coeff = data_model.learn_from_candidates_coeff
+
+    def _init_problem_config(self) -> None:
+        cfg = domain_to_problem_config(self.domain)
+        self._problem_config: ProblemConfig = cfg[0]
+        self._model_pyo: pyo.ConcreteModel = cfg[1]
+
+    @property
+    def input_preprocessing_specs(self):
+        return {}
+
+    def _postprocess_candidate(self, candidate: List) -> pd.DataFrame:
+        """Converts a single candidate to a pandas Dataframe with prediction.
+
+        Args:
+            candidate (List): List containing the features of the candidate.
+
+        Returns:
+            pd.DataFrame: Dataframe with candidate.
+        """
+        keys = [feat.name for feat in self._problem_config.feat_list]
+        df_candidate = pd.DataFrame(
+            data=[candidate],
+            columns=keys,
+        )
+
+        preds = self.predict(df_candidate)
+
+        return pd.concat((df_candidate, preds), axis=1)
+
+    def _add_candidate_as_experiment(self, candidate: pd.DataFrame):
+        gamma = self._learn_from_candidates_coeff
+        # overestimate for minimisation, underestimate for maximisation
+        signs = {
+            output.key: -1 if isinstance(output.objective, MaximizeObjective) else 1
+            for output in self.domain.outputs
+        }
+        as_experiment = candidate.assign(
+            **{
+                key: candidate[f"{key}_pred"]
+                + gamma * signs[key] * candidate[f"{key}_sd"]
+                for key in self.domain.outputs.get_keys()
+            }
+        )
+        self.tell(as_experiment)
+
+    def _ask(self, candidate_count: PositiveInt = 1) -> pd.DataFrame:
+        """Generate a single optimal candidate.
+
+        Args:
+            candidate_count (PositiveInt, optional): Number of candidates to be generated. Defaults to 1.
+
+        Returns:
+            pd.DataFrame: DataFrame with a single candidate (proposed experiment).
+        """
+        candidate_lst = []
+        for _ in range(candidate_count):
+            opt_pyo = PyomoOptimizer(self._problem_config, params=self._solver_params)
+            res = opt_pyo.solve(tree_model=self._enting, model_core=self._model_pyo)
+            candidate = self._postprocess_candidate(res.opt_point)
+            candidate_lst.append(candidate)
+            # only refit if generating multiple
+            if candidate_count > 1:
+                self._add_candidate_as_experiment(candidate)
+
+        return pd.concat(candidate_lst)
+
+    def _fit(self, experiments: pd.DataFrame):
+        input_keys = self.domain.inputs.get_keys()
+        output_keys = self.domain.outputs.get_keys()
+
+        X = experiments[input_keys].to_numpy()
```

Comment:
How is `_enting` handling missing values in `y`? The experiments data frame can contain `nan` for the outputs. You can postprocess them in different ways via ´domain.outputs.preprocess_one_valid_output´, `domain.outputs.preprocess_all_valid_outputs`, and `preprocess_any_valid_output`. Here you should at least use `preprocess_any_valid_output`, to only get the valid measurements and do not have a completely empty row in `y`.